### PR TITLE
[CustomOp] Support to specific extra_cflags and exctra_cuda_flags independently

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

New features 
### PR changes

APIs
### What's New?

+ Support to specific `extra_cflags` and `exctra_cuda_flags` independently.
+ Enhance checking `name` argument in `setup`  API and not allow using `"module"` as suffix in `name` .

### CODE REVIEW

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 1]
```diff
@@ -561,14 +555,14 @@ def load(name,
     Args:
         name(str): generated shared library file name.
         sources(list[str]): custom op source files name with .cc/.cu suffix.
-        extra_cflag(list[str]): additional flags used to compile CPP files. By default
+        extra_cxx_cflags(list[str]): additional flags used to compile CPP files. By default
```
Comment (chenwhql) [inline python/paddle/utils/cpp_extension/cpp_extension.py:558]: `extra_cxx_cflags(list[str], optional)`, others same

File: python/paddle/fluid/tests/custom_op/test_dispatch_jit.py [hunk 2]
```diff
@@ -30,7 +30,8 @@
     name='dispatch_op',
     sources=['dispatch_test_op.cc'],
     extra_include_paths=paddle_includes,  # add for Coverage CI
-    extra_cflags=extra_compile_args,  # add for Coverage CI
+    extra_cxx_cflags=extra_compile_args,
+    extra_cuda_cflags=extra_compile_args,  # add for Coverage CI
```
Comment (chenwhql) [inline python/paddle/fluid/tests/custom_op/test_dispatch_jit.py:34]: only cc test may not need `extra_cuda_cflags`

File: python/paddle/fluid/tests/custom_op/test_multi_out_jit.py [hunk 3]
```diff
@@ -34,7 +34,8 @@
     name='multi_out_jit',
     sources=['multi_out_test_op.cc'],
     extra_include_paths=paddle_includes,  # add for Coverage CI
-    extra_cflags=extra_compile_args,  # add for Coverage CI
+    extra_cxx_cflags=extra_compile_args,  # add for Coverage CI
+    extra_cuda_cflags=extra_compile_args,  # add for Coverage CI
```
Comment (chenwhql) [inline python/paddle/fluid/tests/custom_op/test_multi_out_jit.py:38]: same above

**Review by chenwhql — APPROVED** [review]
LGTM, can polish details in next PR

**Review by zhwesky2010 — APPROVED** [review]
LGTM

File: python/paddle/utils/cpp_extension/cpp_extension.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/utils/cpp_extension/cpp_extension.py:558]: 我在英文文档的PR里会统一修改

File: python/paddle/fluid/tests/custom_op/test_multi_out_jit.py (hunk 3 above)
Comment (Aurelius84) [reply python/paddle/fluid/tests/custom_op/test_multi_out_jit.py:38]: 我在英文文档的PR里会统一修改

File: python/paddle/fluid/tests/custom_op/test_dispatch_jit.py (hunk 2 above)
Comment (Aurelius84) [reply python/paddle/fluid/tests/custom_op/test_dispatch_jit.py:34]: 我在英文文档的PR里会统一修改

**Pull request merged**